### PR TITLE
Allow rerunning flaky tests in unittest.TestCase classes.

### LINKED ISSUE #30: Test classes inheriting TestCase aren't rerun with pytest?
Hi! This seems like a really useful tool! But I'm having some troubles getting it to work with my project.

In the readme there's an example of "Marking a class flaky":

```
@flaky
class TestMultipliers(TestCase):
    def test_flaky_doubler(self):
        value_to_double = 21
        result = get_result_from_flaky_doubler(value_to_double)
        self.assertEqual(result, value_to_double * 2, 'Result doubled incorrectly.')
```

Note how it's inheriting `TestCase`. That's how we are writing our test classes in the project I tried to use `flaky` in.

The problem is that it does not seem to retry the tests if classes are inheriting `TestCase` when using pytest. It only works if it's inheriting `object`, like in the `test_pytest_example.py` module.

Please see my fork for an example of a test case that fails but shouldn't (?)
- Commit: https://github.com/simlun/flaky/commit/6c0c7fcc11de73c00f5430f404b816ecfef9a099
- Failing CI log: https://travis-ci.org/simlun/flaky/jobs/53627102

Best regards
Simon

### PR DESCRIPTION
**Pull request by Jeff-Meadows**
Fixes #30.

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @Jeff-Meadows has signed the CLA. Thanks for the pull request!

File: flaky/flaky_pytest_plugin.py [hunk 1]
```diff
@@ -72,6 +72,14 @@ class FlakyPlugin(_FlakyPlugin):
     max_runs = None
     min_passes = None
 
+    @staticmethod
+    def _get_test_instance(item):
```
Comment (jmoldow) [inline flaky/flaky_pytest_plugin.py]: docstring

File: flaky/flaky_pytest_plugin.py [hunk 2]
```diff
@@ -231,11 +241,24 @@ def __init__(self, plugin, item, func, when):
...
         try:
             self.call(func, plugin)
         finally:
             self.stop = time()
 
+    def _handle_error(self, plugin):
+        # pylint:disable=no-member
+        err = self.excinfo or py.code.ExceptionInfo()
+        # pylint:enable=no-member
+        self.excinfo = None
```
Comment (jmoldow) [inline flaky/flaky_pytest_plugin.py]: Can you also include this line in ``__init__``, so you don't have the no-member lint error?

File: flaky/flaky_pytest_plugin.py [hunk 3]
```diff
@@ -231,11 +241,24 @@ def __init__(self, plugin, item, func, when):
         self.when = when
         self.start = time()
         self._item = item
+        self._want_rerun = []
         try:
             self.call(func, plugin)
         finally:
             self.stop = time()
 
+    def _handle_error(self, plugin):
```
Comment (jmoldow) [inline flaky/flaky_pytest_plugin.py]: docstring

File: flaky/flaky_pytest_plugin.py [hunk 4]
```diff
@@ -251,6 +274,17 @@ def call(self, func, plugin):
...
         try:
             self.result = func()
+            # pytest's unittest plugin for some reason doesn't actually raise
+            # errors. It just adds them to the unittest result. In order to
+            # determine whether or not the test needs to be rerun, this
+            # code looks for the _excinfo attribute set by the plugin.
+            excinfo = getattr(self._item, '_excinfo', None)
+            if (
+                    excinfo is not None and
+                    isinstance(excinfo, list) and
```
Comment (jmoldow) [inline flaky/flaky_pytest_plugin.py]: Doesn't this line make the previous line unnecessary?

You can also have ``excinfo`` default to ``[]``.

**Comment by nzakas** [comment]
@Jeff-Meadows can you please respond to the review comments?

File: flaky/flaky_pytest_plugin.py (hunk 4 above)
Comment (Jeff-Meadows) [inline flaky/flaky_pytest_plugin.py]: It does make it unnecessary.

File: flaky/flaky_pytest_plugin.py (hunk 3 above)
Comment (Jeff-Meadows) [inline flaky/flaky_pytest_plugin.py]: Done.

File: flaky/flaky_pytest_plugin.py (hunk 2 above)
Comment (Jeff-Meadows) [inline flaky/flaky_pytest_plugin.py]: Done.

File: flaky/flaky_pytest_plugin.py (hunk 1 above)
Comment (Jeff-Meadows) [inline flaky/flaky_pytest_plugin.py]: Done.

**Comment by jmoldow** [comment]
:+1: :+1:

**Pull request merged**